### PR TITLE
[BugFix]Fix default warehouse id bug

### PR DESCRIPTION
**Pull request by abc982627271**
Fixes #issue

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 11 / 22 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/lake/StarOSAgent.java|4|12|33.33%|[573, 574, 575, 576, 577, 578, 579, 580]|
|:large_blue_circle:|com/starrocks/warehouse/Cluster.java|5|8|62.50%|[94, 95, 98]|
|:large_blue_circle:|com/starrocks/server/WarehouseManager.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/warehouse/LocalWarehouse.java|1|1|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/lake/StarOSAgent.java [hunk 1]
```diff
@@ -564,4 +565,29 @@ public boolean queryMetaGroupStable(long metaGroupId) {
         }
         return false; // return false if any error happens
     }
+
+    public List<Long> getWorkersByWorkerGroup(List<Long> workerGroupIds) throws UserException {
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/lake/StarOSAgent.java:569]: if allow multiple groups to be listed together, return type should be `List<List<Long>>`, suggest only list single workgroup for now.

File: fe/fe-core/src/main/java/com/starrocks/warehouse/LocalWarehouse.java [hunk 2]
```diff
@@ -41,7 +42,7 @@ public class LocalWarehouse extends Warehouse {
 
     public LocalWarehouse(long id, String name) {
         super(id, name);
-        long clusterId = GlobalStateMgr.getCurrentState().getNextId();
+        long clusterId = DEFAULT_CLUSTER_ID;
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/warehouse/LocalWarehouse.java:45]:
```suggestion
        cluster = new Cluster(DEFAULT_CLUSTER_ID);
```

File: fe/fe-core/src/main/java/com/starrocks/lake/StarOSAgent.java [hunk 3]
```diff
@@ -564,4 +565,30 @@ public boolean queryMetaGroupStable(long metaGroupId) {
...
+        try {
+            List<WorkerGroupDetailInfo> workerGroupDetailInfos = client.
+                    listWorkerGroup(serviceId, Collections.singletonList(workerGroupId), true);
+            for (WorkerGroupDetailInfo detailInfo : workerGroupDetailInfos) {
+                List<WorkerInfo> workerInfos = detailInfo.getWorkersInfoList();
+                for (WorkerInfo workerInfo : workerInfos) {
+                    if (workerToBackend.containsKey(workerInfo.getWorkerId())) {
+                        nodeIds.add(workerToBackend.get(workerInfo.getWorkerId()));
+                    } else {
+                        // for follower fe
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/lake/StarOSAgent.java:581]: is this comment accurate?

File: fe/fe-core/src/main/java/com/starrocks/lake/StarOSAgent.java [hunk 4]
```diff
@@ -564,4 +565,30 @@ public boolean queryMetaGroupStable(long metaGroupId) {
...
+                        String workerAddr = workerInfo.getIpPort();
+                        String[] pair = workerAddr.split(":");
+                        long nodeId = getAvailableBackendId(pair[0], Integer.parseInt(pair[1]));
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/lake/StarOSAgent.java:584]: nodeId may be -1?

File: fe/fe-core/src/main/java/com/starrocks/lake/StarOSAgent.java (hunk 3 above)
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/lake/StarOSAgent.java:581]: `workerToBackend` in leader may also be not complete.
better to update the map at the end.

File: fe/fe-core/src/main/java/com/starrocks/warehouse/Cluster.java [hunk 5]
```diff
@@ -39,7 +42,7 @@ public class Cluster implements Writable {
     private long id;
     @SerializedName(value = "wgid")
     private long workerGroupId;
-    @SerializedName(value = "computeNodeIds")
+
     private Set<Long> computeNodeIds = new HashSet<>();
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/warehouse/Cluster.java:46]: `computeNodeIds` seems useless.

File: fe/fe-core/src/main/java/com/starrocks/warehouse/Cluster.java [hunk 6]
```diff
@@ -74,16 +77,21 @@ public void addNode(long cnId) {
         computeNodeIds.add(cnId);
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/warehouse/Cluster.java:77]: no need to add/drop node when add/drop backend

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 28 / 31 (90.32%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/warehouse/Cluster.java|5|7|71.43%|[76, 79]|
|:large_blue_circle:|com/starrocks/lake/StarOSAgent.java|21|22|95.45%|[579]|
|:large_blue_circle:|com/starrocks/server/WarehouseManager.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/warehouse/LocalWarehouse.java|1|1|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/warehouse/Cluster.java [hunk 7]
```diff
@@ -70,20 +68,16 @@ public void getProcNodeData(BaseProcResult result) {
...
-
     public List<Long> getComputeNodeIds() {
-        return computeNodeIds.stream().collect(Collectors.toList());
+        // ask starMgr for node lists
+        List<Long> nodeIds = new ArrayList<>();
+        try {
+            nodeIds = GlobalStateMgr.getCurrentStarOSAgent().
+                    getWorkersByWorkerGroup(workerGroupId);
+        } catch (UserException e) {
+            LOG.info(e);
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/warehouse/Cluster.java:78]:
```suggestion
            LOG.warn(e);
```

**Review by wyb — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Review by kevincai — APPROVED** [review]

**Pull request merged**